### PR TITLE
Add loading state to pages list on the sidebar

### PR DESCRIPTION
**Pull request by yusufkinatas**
## 🌟 What is the purpose of this PR?

This PR adds a loading state to the pages list on the sidebar to improve UX & clarity.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202787635216887/1202889717609170/f) _(internal)_

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Login to an account with pages
1.  Confirm the loading on the sidebar is visible

## 📹 Demo

https://user-images.githubusercontent.com/39553853/187604404-38667b1e-9164-431a-93b1-53da2a697442.mov

### CODE REVIEW

File: packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list.tsx [hunk 1]
```diff
@@ -65,7 +66,7 @@ export const AccountPageList: FunctionComponent<AccountPageListProps> = ({
   currentPageEntityId,
   accountId,
 }) => {
-  const { data } = useAccountPages(accountId);
+  const { data, loading: pagesLoading } = useAccountPages(accountId);
```
Comment (luisbettencourt) [inline packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list.tsx:69]: I realize this was what the hook returned already but from [this](https://github.com/hashintel/hash/pull/931#discussion_r958837057) comment in another PR we are trying to go with 
`[function, { loading, ...etc }]` instead. If you want I can make this change in that PR when this one gets merged.

**Review by luisbettencourt — APPROVED** [review]
This looks great, thank you @yusufkinatas !

File: packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list.tsx (hunk 1 above)
Comment (yusufkinatas) [reply packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list.tsx:69]: Using an array like that on hook return is more convenient for hooks that return an `action`, not the `data` afaik.

Comment (luisbettencourt) [reply packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list.tsx:69]: Oh yeah you're right. And that's how Apollo does it which was the initial point anyway. Thanks for clearing that up

**Comment by nathggns** [comment]
I don't know if we care *this* deeply, but the lines are not perfectly aligned with where the pages will appear 
image

File: packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list.tsx [hunk 2]
```diff
@@ -120,19 +121,23 @@ export const AccountPageList: FunctionComponent<AccountPageListProps> = ({
...
-        {formattedData.map((node) => renderTree(node, accountId, 0))}
-      </TreeView>
+      {pagesLoading ? (
+        <PagesLoadingState />
+      ) : (
+        <TreeView
+          data-testid="pages-tree"
+          tabIndex={-1}
+          sx={{
+            mx: 0.75,
```
Comment (nathggns) [inline packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list.tsx:131]: This is a margin duplicated between both branches, but also located fairly far apart – someone changing one may not know they need to change both

**Comment by yusufkinatas** [comment]
Improved the alignment & style of the loading. It should be pixel-perfect now.

Screen Shot 2022-09-01 at 10 03 23

**Review by nathggns — APPROVED** [review]

**Pull request merged**